src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp

```diff
@@ -752,10 +753,24 @@ void SolidMechanicsLagrangianFEM::crsApplyTractionBC( real64 const time,
         localIndex const numNodes = faceToNodeMap.sizeOfArray( kf );
         for( localIndex a=0; a<numNodes; ++a )
         {
-          localIndex const dof = blockLocalDofNumber[ faceToNodeMap( kf, a ) ] + component - dofRankOffset;
-          if( dof < 0 || dof >= localRhs.size() )
-            continue;
-          RAJA::atomicAdd< parallelDeviceAtomic >( &localRhs[ dof ], value * faceArea[kf] / numNodes );
+          if( component != -1 )
```

Comment:
I prefer this simple option for this task instead of complexify both the code and the input file. How do you think about this @rrsettgast ?

Reply:
Hi @sytuannguyen :)

Simple is a relative concept: while the code is surely simple to write, it's less simple to read, maintain or understand.
When I read `if( component != -1 )` I am unable to understand what this means without further investigations: I do not consider this code simple for this reason.

Furthermore, this value is directly exposed to the end-user up to the heart of BC settings, so if I want to modify something in the code someday I will have a lot to modify (to the users input files in fact). Worse, I'll need to be aware that there is a constant value here, making refactoring even more dangerous.

Also, the code of this function (and it's not because of you 😉  ) is duplicated (even quadrupled in some sense), so we need to be careful about this in the future.

Maybe/probably it's not to you to fix all of this, but it's worth mentioning.

Reply:
I agree with your points. For instant I will add some comments to clarify this flag. Let's make this BC work first for the actual requirement in the wellbore applications. Then we will optimize it together with the whole solid solver in future PRs.

Reply:
@CusiniM How are you applying pressures to the embedded fractures?

Reply:
Hmm. I have registered a field on the `embeddedSurfaceSubRegion` called `Traction` and that field is updated by a constitutive model (the contact relationship) based on the primary variables. In my case it depends on the pressure and on the fracture aperture. Being a registered field I can also set it as a b.c. in the fieldSpecifications. This is not in develop yet but it is all in https://github.com/GEOSX/GEOSX/pull/1229/files